{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1. annotate vcf file\n",
    "\n",
    "the location of data on genomDK is\n",
    "/home/joco/faststorage/soyadapt_data_analysis_with_start_data/resources/start_data\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "i will work with SNPeff on genomedk\n",
    "for this i will make a seperate projhect folder as the otherfolders are backups of the rest of my project from my laptop+github "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "snpeff/\n",
    "   README.txt\n",
    "   resources/\n",
    "       README.txt\n",
    "       ...\n",
    "   steps/\n",
    "       README.txt\n",
    "       ...\n",
    "   results/\n",
    "       README.txt\n",
    "       ...\n",
    "   plots/\n",
    "       README.txt\n",
    "       ...\n",
    "   workflow/\n",
    "       README.txt\n",
    "       notebooks\n",
    "       envs\n",
    "         environment.yml"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "conda activate\n",
    "\n",
    "conda create -n snpeffenv \n",
    "\n",
    "Download latest version\n",
    "wget https://snpeff.blob.core.windows.net/versions/snpEff_latest_core.zip\n",
    "\n",
    "Unzip file\n",
    "unzip snpEff_latest_core.zip\n",
    "\n",
    "i had to update to java 11 on the environment only\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "shellscript"
    }
   },
   "outputs": [],
   "source": [
    "#conda install -c conda-forge openjdk=11\n",
    "java -version\n",
    "\n",
    "#local !!! You have to be in the unzipped /snpEff file to run the following command! !!\n",
    "\n",
    "#on genome dk go to \n",
    "cd /home/joco/miniconda3/pkgs/snpeff-5.1-hdfd78af_0/share/snpeff-5.1-0/\n",
    "#found its location after downlading via conda with\n",
    "# find / -name snpEff.jar 2>/dev/null\n",
    "\n",
    "java -jar snpEff.jar databases | grep -i glycine_max\n",
    "\n",
    "Glycine_max                                                 \tGlycine_max                                                 \t          \t                              \t[https://snpeff.blob.core.windows.net/databases/v5_1/snpEff_v5_1_Glycine_max.zip, https://snpeff.blob.core.windows.net/databases/v5_0/snpEff_v5_0_Glycine_max.zip]\n",
    "\n",
    " java -jar snpEff.jar download Glycine_max"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "vscode": {
     "languageId": "shellscript"
    }
   },
   "source": [
    "collect the files\n",
    "\n",
    "1. path to reference genome file (fasta) - the same as used for VCF construction\n",
    "\n",
    "2. path to Reference genome gene annotation (gff) :\n",
    "\n",
    "3. path to output directory\n",
    "\n",
    "4. path to vcf input\n",
    "5. the downloaded database data\n",
    "\n",
    "\n",
    "\n",
    "in genomeDK:\n",
    "1. ref=/home/joco/faststorage/snpeff_analysis/resources/Wm82.a2.v1.genome.fasta\n",
    "2. ann=/home/joco/faststorage/snpeff_analysis/resources/Glycine_max.Glycine_max_v2.1.56.chr.gff3.gz\n",
    "\n",
    "3. out=/home/joco/faststorage/snpeff_analysis/resources/generated_data/\n",
    "4. wgs_maf5=/home/joco/faststorage/snpeff_analysis/resources/start_data/wgs_maf5.vcf.gz\n",
    "5. snpeff_data=/home/joco/miniconda3/pkgs/snpeff-5.1-hdfd78af_0/share/snpeff-5.1-0/data/Glycine_max\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "i checked to see if the chromsomes have the same name in my vcf file as the .fasta and .gff3\n",
    "\n",
    "\n",
    "\n",
    "zcat Glycine_max.Glycine_max_v2.1.56.chr.gff3.gz | cut -f1 | sort | uniq\n",
    "\n",
    "fasta file\n",
    "\n",
    "zcat your_file.fasta.gz | grep \"^>\" | cut -d \" \" -f 1 | sed 's/>//'\n",
    "\n",
    "so maybe not the same \n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# run snpeff with database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "shellscript"
    }
   },
   "outputs": [],
   "source": [
    "in:\n",
    "/home/joco/miniconda3/pkgs/snpeff-5.1-hdfd78af_0/share/snpeff-5.1-0/\n",
    "\n",
    "export wgs_maf5=/home/joco/miniconda3/pkgs/snpeff-5.1-hdfd78af_0/share/snpeff-5.1-0/data/wgs_maf5.vcf.gz\n",
    "\n",
    "java -Xmx8g -jar snpEff.jar -v Glycine_max  $wgs_maf5 > wgs_maf5.ann.vcf.gz\n",
    "# i will make one more with stats!\n",
    "java -Xmx8g -jar snpEff.jar -v -stats ex1.html Glycine_max $wgs_maf5 > 2_wgs_maf5.ann.vcf.gzann.vcf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "shellscript"
    }
   },
   "outputs": [],
   "source": [
    "#great i now have a file:\n",
    "/home/joco/miniconda3/pkgs/snpeff-5.1-hdfd78af_0/share/snpeff-5.1-0/wgs_maf5.ann.vcf.gz\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "i now want to compare the windows with differing values of signatures of selection in the list of genes against the rest of the genome\n",
    "\n",
    "i need a Gene List: Ensure you have a list of genes you want to compare against the rest of the genome. This list could be in a simple text file, with one gene symbol per line.\n",
    "\n",
    "Gene Coordinates: Obtain the genomic coordinates (start and end positions) of each gene in your gene list. \n",
    "\n",
    "\n",
    "    Calculate Average Signature Value: For each gene in your list, calculate the average value of signatures of selection across the windows that overlap with the gene's coordinates. You can write a script or use bioinformatics tools to do this.\n",
    "\n",
    "    Calculate Average Value for Rest of Genome: Calculate the average value of signatures of selection for all the windows in the rest of the genome (i.e., excluding the windows that overlap with the genes in your list).\n",
    "\n",
    "    Statistical Comparison: Perform a statistical comparison, such as a t-test or Wilcoxon rank-sum test, to determine if the average signature value in your gene list is significantly different from the average value in the rest of the genome.\n",
    "\n",
    "    Multiple Testing Correction (Optional): If you are comparing against a large number of windows, consider applying multiple testing correction methods, such as Bonferroni or False Discovery Rate (FDR) correction, to account for the increased risk of false positives.\n",
    "\n",
    "    Visualization: You can create visualizations to compare the average signature values between your gene list and the rest of the genome. Box plots, violin plots, or bar plots can be useful for this purpose.\n",
    "\n",
    "Keep in mind that bioinformatics analysis can be complex, and the specific tools and scripts you'll use may depend on the format of your data and the programming language you are comfortable with. If you are not familiar with bioinformatics analyses, it might be helpful to seek assistance from a bioinformatician or collaborator with expertise in this area. Additionally, various bioinformatics software tools like R packages or Python libraries can aid in data manipulation, analysis, and visualization during this process."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "bcftools view -h wgs_maf5.ann.vcf.gz   # View the VCF header\n",
    "#checked out the annotation format\n",
    "bcftools view wgs_maf5.ann.vcf.gz | less\n",
    "\n",
    "\n",
    "i have a list of genes to try out\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Glyma01g02520\n",
    "Glyma01g02530\n",
    "Glyma01g02880\n",
    "Glyma01g05960\n",
    "Glyma01g07070\n",
    "Glyma01g08130\n",
    "Glyma01g08150\n",
    "Glyma01g11670\n",
    "Glyma01g21710\n",
    "Glyma01g22260\n",
    "Glyma01g36730\n",
    "Glyma01g36810\n",
    "Glyma01g37190\n",
    "Glyma01g37470\n",
    "Glyma01g39490\n",
    "Glyma01g39520\n",
    "Glyma01g39950\n",
    "Glyma01g41460\n",
    "Glyma01g43360\n",
    "Glyma02g00830\n",
    "Glyma02g00980\n",
    "Glyma02g01540\n",
    "Glyma02g01760\n",
    "Glyma02g04710\n",
    "Glyma02g05100\n",
    "Glyma02g07650\n",
    "Glyma02g09600\n",
    "Glyma02g09860\n",
    "Glyma02g10880\n",
    "Glyma02g11060\n",
    "Glyma02g12080\n",
    "Glyma02g12930\n",
    "Glyma02g13400\n",
    "Glyma02g13420\n",
    "Glyma02g14880\n",
    "Glyma02g15850\n",
    "Glyma02g17110\n",
    "Glyma02g18610\n",
    "Glyma02g29380\n",
    "Glyma02g33040\n",
    "Glyma02g36210\n",
    "Glyma02g37120\n",
    "Glyma02g37880\n",
    "Glyma02g38090\n",
    "Glyma02g43540\n",
    "Glyma02g43760\n",
    "Glyma02g45730\n",
    "Glyma02g46680\n",
    "Glyma02g46970\n",
    "Glyma03g00580\n",
    "Glyma03g01030\n",
    "Glyma03g02200\n",
    "Glyma03g02210\n",
    "Glyma03g22710\n",
    "Glyma03g23260\n",
    "Glyma03g27010\n",
    "Glyma03g31670\n",
    "Glyma03g33470\n",
    "Glyma03g33490\n",
    "Glyma03g35250\n",
    "Glyma03g35610\n",
    "Glyma03g36970\n",
    "Glyma03g37840\n",
    "Glyma03g38320\n",
    "Glyma03g38390\n",
    "Glyma03g38620\n",
    "Glyma03g38670\n",
    "Glyma03g39910\n",
    "Glyma03g40840\n",
    "Glyma03g42220\n",
    "Glyma03g42260\n",
    "Glyma04g00340\n",
    "Glyma04g01400\n",
    "Glyma04g02420\n",
    "Glyma04g02980\n",
    "Glyma04g04170\n",
    "Glyma04g05280\n",
    "Glyma04g06240\n",
    "Glyma04g07540\n",
    "Glyma04g07900\n",
    "Glyma04g08680\n",
    "Glyma04g09820\n",
    "Glyma04g10820\n",
    "Glyma04g11010\n",
    "Glyma04g14840\n",
    "Glyma04g31800\n",
    "Glyma04g31810\n",
    "Glyma04g32540\n",
    "Glyma04g33110\n",
    "Glyma04g33410\n",
    "Glyma04g35810\n",
    "Glyma04g36620\n",
    "Glyma04g36630\n",
    "Glyma04g37290\n",
    "Glyma04g37900\n",
    "Glyma04g38060\n",
    "Glyma04g38860\n",
    "Glyma04g40640\n",
    "Glyma04g42410\n",
    "Glyma04g42420\n",
    "Glyma04g43620\n",
    "Glyma04g43640\n",
    "Glyma05g00880\n",
    "Glyma05g00970\n",
    "Glyma05g01510\n",
    "Glyma05g03660\n",
    "Glyma05g07350\n",
    "Glyma05g07380\n",
    "Glyma05g13890\n",
    "Glyma05g18170\n",
    "Glyma05g22250\n",
    "Glyma05g22320\n",
    "Glyma05g25200\n",
    "Glyma05g26150\n",
    "Glyma05g26460\n",
    "Glyma05g28130\n",
    "Glyma05g28140\n",
    "Glyma05g29090\n",
    "Glyma05g29590\n",
    "Glyma05g31680\n",
    "Glyma05g32680\n",
    "Glyma05g34030\n",
    "Glyma05g34280\n",
    "Glyma05g34530\n",
    "Glyma05g35280\n",
    "Glyma05g37070\n",
    "Glyma06g00400\n",
    "Glyma06g01430\n",
    "Glyma06g02470\n",
    "Glyma06g02990\n",
    "Glyma06g04350\n",
    "Glyma06g06300\n",
    "Glyma06g08790\n",
    "Glyma06g09890\n",
    "Glyma06g10020\n",
    "Glyma06g10650\n",
    "Glyma06g10830\n",
    "Glyma06g12380\n",
    "Glyma06g12390\n",
    "Glyma06g14150\n",
    "Glyma06g17010\n",
    "Glyma06g17170\n",
    "Glyma06g17780\n",
    "Glyma06g18290\n",
    "Glyma06g18300\n",
    "Glyma06g20950\n",
    "Glyma06g21120\n",
    "Glyma06g21900\n",
    "Glyma06g22650\n",
    "Glyma06g26590\n",
    "Glyma06g37080\n",
    "Glyma06g37220\n",
    "Glyma06g38600\n",
    "Glyma06g47220\n",
    "Glyma06g48270\n",
    "Glyma07g01600\n",
    "Glyma07g04180\n",
    "Glyma07g05410\n",
    "Glyma07g05530\n",
    "Glyma07g06420\n",
    "Glyma07g08820\n",
    "Glyma07g08830\n",
    "Glyma07g08890\n",
    "Glyma07g08920\n",
    "Glyma07g09800\n",
    "Glyma07g09990\n",
    "Glyma07g21160\n",
    "Glyma07g25970\n",
    "Glyma07g30040\n",
    "Glyma07g31880\n",
    "Glyma07g33600\n",
    "Glyma07g35240\n",
    "Glyma07g35690\n",
    "Glyma07g37640\n",
    "Glyma07g37770\n",
    "Glyma07g39350\n",
    "Glyma08g00330\n",
    "Glyma08g02490\n",
    "Glyma08g04440\n",
    "Glyma08g05130\n",
    "Glyma08g05390\n",
    "Glyma08g05650\n",
    "Glyma08g06980\n",
    "Glyma08g07260\n",
    "Glyma08g08220\n",
    "Glyma08g09090\n",
    "Glyma08g09380\n",
    "Glyma08g11110\n",
    "Glyma08g11120\n",
    "Glyma08g12230\n",
    "Glyma08g12730\n",
    "Glyma08g14930\n",
    "Glyma08g15700\n",
    "Glyma08g17630\n",
    "Glyma08g19590\n",
    "Glyma08g21110\n",
    "Glyma08g22400\n",
    "Glyma08g24340\n",
    "Glyma08g24550\n",
    "Glyma08g24590\n",
    "Glyma08g27670\n",
    "Glyma08g27680\n",
    "Glyma08g28370\n",
    "Glyma08g28470\n",
    "Glyma08g36380\n",
    "Glyma08g42300\n",
    "Glyma08g43760\n",
    "Glyma08g44140\n",
    "Glyma08g44640\n",
    "Glyma08g44650\n",
    "Glyma08g47810\n",
    "Glyma08g47820\n",
    "Glyma08g47900\n",
    "Glyma09g01000\n",
    "Glyma09g01650\n",
    "Glyma09g03990\n",
    "Glyma09g04970\n",
    "Glyma09g05080\n",
    "Glyma09g05090\n",
    "Glyma09g06220\n",
    "Glyma09g06750\n",
    "Glyma09g07120\n",
    "Glyma09g07240\n",
    "Glyma09g07770\n",
    "Glyma09g11600\n",
    "Glyma09g17220\n",
    "Glyma09g18790\n",
    "Glyma09g20280\n",
    "Glyma09g26550\n",
    "Glyma09g27450\n",
    "Glyma09g28670\n",
    "Glyma09g29030\n",
    "Glyma09g31770\n",
    "Glyma09g32010\n",
    "Glyma09g37170\n",
    "Glyma09g38290\n",
    "Glyma09g40230\n",
    "Glyma09g42060\n",
    "Glyma09g42120\n",
    "Glyma09g42140\n",
    "Glyma10g01580\n",
    "Glyma10g01830\n",
    "Glyma10g02690\n",
    "Glyma10g03910\n",
    "Glyma10g05520\n",
    "Glyma10g05610\n",
    "Glyma10g08340\n",
    "Glyma10g08370\n",
    "Glyma10g08710\n",
    "Glyma10g22390\n",
    "Glyma10g23370\n",
    "Glyma10g23420\n",
    "Glyma10g28170\n",
    "Glyma10g28290\n",
    "Glyma10g29690\n",
    "Glyma10g31190\n",
    "Glyma10g31450\n",
    "Glyma10g32390\n",
    "Glyma10g33550\n",
    "Glyma10g34760\n",
    "Glyma10g35350\n",
    "Glyma10g36600\n",
    "Glyma10g36820\n",
    "Glyma10g38580\n",
    "Glyma10g43190\n",
    "Glyma11g02110\n",
    "Glyma11g03950\n",
    "Glyma11g03960\n",
    "Glyma11g05340\n",
    "Glyma11g05720\n",
    "Glyma11g05760\n",
    "Glyma11g07150\n",
    "Glyma11g07820\n",
    "Glyma11g08090\n",
    "Glyma11g09700\n",
    "Glyma11g10670\n",
    "Glyma11g13210\n",
    "Glyma11g13220\n",
    "Glyma11g13490\n",
    "Glyma11g14490\n",
    "Glyma11g15560\n",
    "Glyma11g15580\n",
    "Glyma11g15650\n",
    "Glyma11g16110\n",
    "Glyma11g18190\n",
    "Glyma11g20200\n",
    "Glyma11g29610\n",
    "Glyma11g29860\n",
    "Glyma11g35270\n",
    "Glyma11g36890\n",
    "Glyma12g02980\n",
    "Glyma12g03700\n",
    "Glyma12g04670\n",
    "Glyma12g05250\n",
    "Glyma12g05490\n",
    "Glyma12g06410\n",
    "Glyma12g07800\n",
    "Glyma12g07860\n",
    "Glyma12g10320\n",
    "Glyma12g13560\n",
    "Glyma12g17720\n",
    "Glyma12g25240\n",
    "Glyma12g30940\n",
    "Glyma12g30980\n",
    "Glyma12g34240\n",
    "Glyma12g34510\n",
    "Glyma12g35320\n",
    "Glyma13g00510\n",
    "Glyma13g00860\n",
    "Glyma13g00920\n",
    "Glyma13g01290\n",
    "Glyma13g01810\n",
    "Glyma13g02310\n",
    "Glyma13g03880\n",
    "Glyma13g05480\n",
    "Glyma13g06730\n",
    "Glyma13g07030\n",
    "Glyma13g09660\n",
    "Glyma13g10690\n",
    "Glyma13g16700\n",
    "Glyma13g16850\n",
    "Glyma13g19870\n",
    "Glyma13g22030\n",
    "Glyma13g22060\n",
    "Glyma13g24530\n",
    "Glyma13g24590\n",
    "Glyma13g26820\n",
    "Glyma13g27770\n",
    "Glyma13g27780\n",
    "Glyma13g27790\n",
    "Glyma13g29510\n",
    "Glyma13g30330\n",
    "Glyma13g31050\n",
    "Glyma13g31090\n",
    "Glyma13g33040\n",
    "Glyma13g35190\n",
    "Glyma13g35980\n",
    "Glyma13g36310\n",
    "Glyma13g39340\n",
    "Glyma13g39360\n",
    "Glyma13g40470\n",
    "Glyma13g41140\n",
    "Glyma13g42060\n",
    "Glyma13g42660\n",
    "Glyma14g03100\n",
    "Glyma14g05150\n",
    "Glyma14g05430\n",
    "Glyma14g06480\n",
    "Glyma14g07960\n",
    "Glyma14g10530\n",
    "Glyma14g17260\n",
    "Glyma14g21260\n",
    "Glyma14g24590\n",
    "Glyma14g33810\n",
    "Glyma14g35020\n",
    "Glyma14g35410\n",
    "Glyma14g36100\n",
    "Glyma14g36220\n",
    "Glyma14g37280\n",
    "Glyma15g02770\n",
    "Glyma15g03330\n",
    "Glyma15g04260\n",
    "Glyma15g04930\n",
    "Glyma15g05440\n",
    "Glyma15g06300\n",
    "Glyma15g06310\n",
    "Glyma15g06470\n",
    "Glyma15g06490\n",
    "Glyma15g08270\n",
    "Glyma15g08320\n",
    "Glyma15g08860\n",
    "Glyma15g09500\n",
    "Glyma15g11850\n",
    "Glyma15g12570\n",
    "Glyma15g14980\n",
    "Glyma15g15730\n",
    "Glyma15g15890\n",
    "Glyma15g17480\n",
    "Glyma15g18010\n",
    "Glyma15g18050\n",
    "Glyma15g18450\n",
    "Glyma15g19390\n",
    "Glyma15g23400\n",
    "Glyma15g23610\n",
    "Glyma15g29870\n",
    "Glyma15g35080\n",
    "Glyma15g37830\n",
    "Glyma16g00890\n",
    "Glyma16g01980\n",
    "Glyma16g02050\n",
    "Glyma16g03030\n",
    "Glyma16g04830\n",
    "Glyma16g04840\n",
    "Glyma16g05110\n",
    "Glyma16g08860\n",
    "Glyma16g13070\n",
    "Glyma16g17450\n",
    "Glyma16g26660\n",
    "Glyma16g26690\n",
    "Glyma16g32080\n",
    "Glyma16g32540\n",
    "Glyma17g01370\n",
    "Glyma17g02860\n",
    "Glyma17g03010\n",
    "Glyma17g03960\n",
    "Glyma17g05860\n",
    "Glyma17g05990\n",
    "Glyma17g06640\n",
    "Glyma17g06950\n",
    "Glyma17g07000\n",
    "Glyma17g07420\n",
    "Glyma17g08890\n",
    "Glyma17g10370\n",
    "Glyma17g10920\n",
    "Glyma17g11040\n",
    "Glyma17g14190\n",
    "Glyma17g17520\n",
    "Glyma17g17790\n",
    "Glyma17g18640\n",
    "Glyma17g29720\n",
    "Glyma17g34980\n",
    "Glyma17g37050\n",
    "Glyma18g00800\n",
    "Glyma18g03130\n",
    "Glyma18g06460\n",
    "Glyma18g07770\n",
    "Glyma18g08620\n",
    "Glyma18g09060\n",
    "Glyma18g11180\n",
    "Glyma18g11400\n",
    "Glyma18g12590\n",
    "Glyma18g33910\n",
    "Glyma18g45780\n",
    "Glyma18g49520\n",
    "Glyma18g50900\n",
    "Glyma18g50910\n",
    "Glyma18g51320\n",
    "Glyma18g53610\n",
    "Glyma18g53670\n",
    "Glyma18g53680\n",
    "Glyma18g53690\n",
    "Glyma19g02710\n",
    "Glyma19g04320\n",
    "Glyma19g05170\n",
    "Glyma19g20090\n",
    "Glyma19g27950\n",
    "Glyma19g28390\n",
    "Glyma19g28400\n",
    "Glyma19g29610\n",
    "Glyma19g30230\n",
    "Glyma19g34470\n",
    "Glyma19g36200\n",
    "Glyma19g36220\n",
    "Glyma19g37890\n",
    "Glyma19g37910\n",
    "Glyma19g38230\n",
    "Glyma19g39620\n",
    "Glyma19g40430\n",
    "Glyma19g40980\n",
    "Glyma19g41210\n",
    "Glyma19g41260\n",
    "Glyma19g42460\n",
    "Glyma19g43540\n",
    "Glyma19g44970\n",
    "Glyma19g45030\n",
    "Glyma20g00350\n",
    "Glyma20g00400\n",
    "Glyma20g01130\n",
    "Glyma20g03120\n",
    "Glyma20g04600\n",
    "Glyma20g10600\n",
    "Glyma20g16880\n",
    "Glyma20g19600\n",
    "Glyma20g22160\n",
    "Glyma20g22280\n",
    "Glyma20g23700\n",
    "Glyma20g24220\n",
    "Glyma20g29250\n",
    "Glyma20g30980\n",
    "Glyma20g32160\n",
    "Glyma20g32730\n",
    "Glyma20g34050\n",
    "Glyma20g35220\n",
    "Glyma20g36110\n",
    "Glyma20g36330\n",
    "Glyma20g37620\n",
    "Glyma20g39140"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "shellscript"
    }
   },
   "outputs": [],
   "source": [
    "# something java -jar snpEff.jar  floweringgenes > flowering_gene_positions.txt\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "shellscript"
    }
   },
   "outputs": [],
   "source": [
    "bcftools query -f '%CHROM\\t%POS\\t%INFO/ANN\\n' wgs_maf5.ann.vcf.gz > output.txt\n",
    "#im looking at ANN\n",
    "# "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##SnpEffVersion=\"5.1 (build 2022-01-21 06:23), by Pablo Cingolani\"\n",
    "##SnpEffCmd=\"SnpEff  Glycine_max /home/joco/miniconda3/pkgs/snpeff-5.1-hdfd78af_0/share/snpeff-5.1-0/data/wgs_maf5.vcf.gz \"\n",
    "##INFO=<ID=ANN,Number=.,Type=String,Description=\"Functional annotations: 'Allele | Annotation | Annotation_Impact | Gene_Name | Gene_ID | Feature_Type | Feature_ID | Transcript_BioType | Rank | HGVS.c | HGVS.p | cDNA.pos / cDNA.length | CDS.pos / CDS.length | AA.pos / AA.length | Distance | ERRORS / WARNINGS / INFO'\">\n",
    "##INFO=<ID=LOF,Number=.,Type=String,Description=\"Predicted loss of function effects for this variant. Format: 'Gene_Name | Gene_ID | Number_of_transcripts_in_gene | Percent_of_transcripts_affected'\">\n",
    "##INFO=<ID=NMD,Number=.,Type=String,Description=\"Predicted nonsense mediated decay effects for this variant. Format: 'Gene_Name | Gene_ID | Number_of_transcripts_in_gene | Percent_of_transcripts_affected'\">\n",
    "\n",
    "1\t27\tG|intergenic_region|MODIFIER|CHR_START-GLYMA_01G000100|CHR_START-GLYMA_01G000100|intergenic_region|CHR_START-GLYMA_01G000100|||n.27A>G||||||\n",
    "so $4\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "shellscript"
    }
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "shellscript"
    }
   },
   "outputs": [],
   "source": [
    "#i moved loval to \n",
    "/Users/josephinelovemachine/soyadapt_data_analysis_with_start_data/resources/generated_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "shellscript"
    }
   },
   "outputs": [],
   "source": [
    "sed 's/|/\\t/g' output.txt | awk -F '\\t' '{print $1, $2, $4, $5, $6, $7}' | awk -F '\\t' '{print $1, $2, tolower($4), tolower($5),tolower($6), tolower($7)}' > extracted_genes.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "notes\n",
    "scans for quality\n",
    "stats?\n",
    "https://github.com/alexherbig/MultiVCFAnalyzer#usage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "ppp_env_x86",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  },
  "vscode": {
   "interpreter": {
    "hash": "fd125518df18430d0936e52ecb7291e22b91211acfd52c6d5373b9b07bdacec8"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
